packages/phpstan-rules/tests/Rules/CheckTypehintCallerTypeRule/CheckTypehintCallerTypeRuleTest.php

```diff
@@ -6,12 +6,26 @@
 
 use Iterator;
 use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\Variable;
+use PHPStan\Analyser\Scope;
 use PHPStan\Rules\Rule;
+use PHPStan\Type\ThisType;
 use Symplify\PHPStanExtensions\Testing\AbstractServiceAwareRuleTestCase;
 use Symplify\PHPStanRules\Rules\CheckTypehintCallerTypeRule;
 
 final class CheckTypehintCallerTypeRuleTest extends AbstractServiceAwareRuleTestCase
 {
+    public function testProcessMethodCallNotHasParent(): void
+    {
+        $scope = $this->createMock(Scope::class);
+        $thisType = $this->createMock(ThisType::class);
+
+        $scope->method('getType')
+            ->willReturn($thisType);
+
+        $this->assertEmpty($this->getRule()->process(new MethodCall(new Variable('this'), 'isCheck'), $scope));
```

Comment:
@TomasVotruba I can reproduce with simulate pass MethodCall instance directly to `$this->getRule()->process()` in the tests.

Is this ok now?

Reply:
That's not compatible with other test fixture. We'll need the exact code that fails on Rector, as other test fixtures have.

Reply:
Ok, I will try find the exact code that cause it.